controllers/base.go

```diff
@@ -163,6 +165,57 @@ func (bmrc *baseMachineReconcileContext) releaseHardware() error {
 		return fmt.Errorf("patching Hardware object: %w", err)
 	}
 
+	if err := bmrc.powerOffHardware(hardware); err != nil {
+		return fmt.Errorf("failed to power off hardware: %w", err)
+	}
+
+	return nil
+}
+
+// powerOffHardware uses hardware BMCRef to create a job to power off the machine.
+func (bmrc *baseMachineReconcileContext) powerOffHardware(hardware *tinkv1.Hardware) error {
+	if hardware.Spec.BMCRef == nil {
+		bmrc.log.Info("Skipping hardware power off", "BMCRef", hardware.Spec.BMCRef, "Hardware", hardware.Name)
```

Comment:
We should log _why_. We should also make 

```suggestion
		bmrc.log.Info("Hardware BMC reference not present; skipping power off", "BMCRef", hardware.Spec.BMCRef, "Hardware", hardware.Name)
```